operationsbus/handlers.go

```diff
@@ -96,18 +96,27 @@ func NewQosErrorHandler(errHandler ErrorHandlerFunc) shuttle.HandlerFunc {
 		err := errHandler.Handle(ctx, settler, message)
 		t := time.Now()
 		elapsed := t.Sub(start)
-		logger.Info("QoS: Operation started at: " + start.String())
-		logger.Info("QoS: Operation processed at: " + t.String())
-		logger.Info("QoS: Operation took " + elapsed.String() + " to process.")
+		logger.Info("QoS: Operation started at: " + start.String() + ". QoS: Operation processed at: " + t.String() + ". QoS: Operation took " + elapsed.String() + " to process.")
 
 		if err != nil {
-			logger.Error("QoS: Error ocurred in previousHandler: " + err.Error())
+			logger.With(
```

Comment:
Can we customize these fields, or at the very least pass in our own logger? Looks like this message will go to the same table as the regular operation logs.

Reply:
ah wait I see the function below. I think these should be combined. How do I know when I'm writing the code if I'm going to have an error or not?

There should just be 1 QoS handler, and it handles both success and error cases

Reply:
Added option to pass your own logger into the QoS handler.

I also added the AsyncError, which has information such as the `OriginalError`, `Message`, `RetryAfter`, and `ErrorCode`, like you explained you needed. I also found the [`async_result.go`](https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/guardrails/server/worker/base/async_result.go) and tried to mimic some of its behavior.

As for the two functions, I'm leaving them separate so users can choose if they want to gather error information (which they should) or not, but I don't want to force them on to one, especially since the original library doesn't use handlers that return an error, so I want to maintain that compatibility if possible.